utils/fasm/src/fasm.cpp

```diff
@@ -138,8 +132,51 @@ std::string FasmWriterVisitor::build_clb_prefix(const t_pb_graph_node* pb_graph_
               fasm_prefix_unsplit.c_str(), pb_type->num_pb);
   }
 
-  return clb_prefix + fasm_prefix.at(pb_graph_node->placement_index) + ".";
+  return fasm_prefix.at(pb_graph_node->placement_index) + ".";
+}
+
+std::string FasmWriterVisitor::build_clb_prefix(const t_pb *pb, const t_pb_graph_node* pb_graph_node) const {
+  std::string clb_prefix = "";
+
+  const t_pb *pb_for_graph_node = nullptr;
+
+  // If not t_pb, mode_index is always 0.
```

Comment:
Lines 145-161 feel like a common algorithm that might be used in multiple places?

Reply:
I don't think any other piece of code needs to do this.  I looked and didn't find any examples.